src/coord/category.rs

```diff
@@ -0,0 +1,236 @@
+use std::fmt;
+use std::ops::Range;
+use std::rc::Rc;
+
+use super::{AsRangedCoord, Ranged};
+
+/// The category coordinate
+pub struct Category<T: PartialEq> {
+    name: String,
+    elements: Rc<Vec<T>>,
+}
+
+/// The category element reference (tick).
+pub struct CategoryElementRef<T: PartialEq> {
+    inner: Rc<Vec<T>>,
+    // i32 type is required for the empty ref (having -1 value)
+    idx: i32,
+}
+
+/// The category elements range.
+pub struct CategoryElementsRange<T: PartialEq>(CategoryElementRef<T>, CategoryElementRef<T>);
```

Comment:
You're right, the initial implementation didn't use reference counting, so I had to use the separate structure to avoid cloning the original vector. But now it's not an issue, and it can be optimized.